### PR TITLE
Fix crash in XAML design mode

### PR DESCRIPTION
**Pull request by MikiraSora**
Fix issue : https://github.com/jayhf/OpenTkControl/issues/6
and another bug when control want to generate a opengl buffer in design mode:

### CODE REVIEW

File: OpenTkControl/OpenTkControlBase.xaml.cs [hunk 1]
```diff
@@ -334,8 +336,15 @@ public Task RequestRepaint()
...
         protected virtual void OnLoaded(object sender, RoutedEventArgs args)
         {
-            _windowInfo = Utilities.CreateWindowsWindowInfo(
-                new WindowInteropHelper(Window.GetWindow(this)).Handle);
+            try
+            {
+                _windowInfo = Utilities.CreateWindowsWindowInfo(
+                    new WindowInteropHelper(Window.GetWindow(this)).Handle);
+            }
+            catch
```
Comment (jayhf) [inline OpenTkControl/OpenTkControlBase.xaml.cs]: Why is this necessary? Can't you just add a design mode check?

File: OpenTkControl/OpenTkControlBase.xaml [hunk 2]
```diff
@@ -1,13 +1,18 @@
-﻿<UserControl x:Class="OpenTkControl.OpenTkControlBase"
-             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
-             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
-             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
-             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
-             mc:Ignorable="d" 
-             d:DesignHeight="300" d:DesignWidth="300">
-    <Image x:Name="Image" Stretch="Fill" RenderTransformOrigin="0.5,0.5">
+﻿<UserControl
```
Comment (jayhf) [inline OpenTkControl/OpenTkControlBase.xaml]: Please don't make unnecessary changes. They just clutter git history

File: OpenTkControl/packages.config [hunk 3]
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="OpenTK" version="2.0.0" />
+  <package id="OpenTK" version="2.0.0" targetFramework="net45" />
```
Comment (jayhf) [inline OpenTkControl/packages.config]: Why?

File: OpenTkControl/OpenTkControlBase.xaml.cs [hunk 4]
```diff
@@ -305,6 +305,8 @@ public Task RequestRepaint()
             return tcs.Task;
         }
 
+        protected bool IsDesignMode => DesignerProperties.GetIsInDesignMode(this);
```
Comment (jayhf) [inline OpenTkControl/OpenTkControlBase.xaml.cs]: I think inlining this would be more clear

File: OpenTkControl/ThreadOpenTkControl.cs [hunk 5]
```diff
@@ -89,6 +90,12 @@ private void OnIsVisibleChanged(object sender, DependencyPropertyChangedEventArg
         /// <param name="boxedToken"></param>
         private void RenderThread(object boxedToken)
         {
+#if DEBUG
+            //We needn't call render() for avoiding crash by calling OpenGL API methods.
```
Comment (jayhf) [inline OpenTkControl/ThreadOpenTkControl.cs]: Grammar
```suggestion
            // Don't render in design mode to prevent errors from calling OpenGL API methods.
```

**Review by jayhf — CHANGES_REQUESTED** [review]
Thanks for looking into this issue. I made a few suggestions.

File: OpenTkControl/packages.config (hunk 3 above)
Comment (MikiraSora) [reply OpenTkControl/packages.config]: this project has been reference by my other projects and VS modify it automatically. fixed

File: OpenTkControl/OpenTkControlBase.xaml (hunk 2 above)
Comment (MikiraSora) [reply OpenTkControl/OpenTkControlBase.xaml]: Sorry it has been formatted automatically by VS plugin XAML Fomatter.

File: OpenTkControl/OpenTkControlBase.xaml.cs (hunk 1 above)
Comment (MikiraSora) [reply OpenTkControl/OpenTkControlBase.xaml.cs]: modified.

File: OpenTkControl/OpenTkControlBase.xaml.cs (hunk 4 above)
Comment (MikiraSora) [reply OpenTkControl/OpenTkControlBase.xaml.cs]: modified

**Comment by MikiraSora** [comment]
In fact I notice glReadPixels spend most time on copying buffer into bitmap though CPU. I think it could use WGL_NV_DX_interop and share with DirectX surface.

**Comment by jayhf** [comment]
The changes look great. I just want to test them before I approve and I probably won't have time until the weekend. You're right that the current CPU copying method isn't efficient and that interop function looks like it could really improve things. I didn't know it existed. I'm not actively developing this library any more, but you're welcome to integrate and PR it if you want.

**Comment by MikiraSora** [comment]
OK I had implemented this opengl extension and release a example.
https://github.com/MikiraSora/OpenTkControl/releases/tag/v0.1.0

I will take a test and format code before raising new PR.

But this is temporary solution because the team of OpenTK dev also want to improve their control library though this extension and WGL_NV_DX_interop2.

**Review by jayhf — APPROVED** [review]
Thanks again for the changes. And sorry I forgot about this for so long. Feel free to open a PR that adds `WGL_NV_DX_interop2` support if that's still something you're interested in.
